### ISSUE TITLE
CNTK fails in building  the model with the layer`ConvLSTM2D`in some cases when `kernel_size` is greater than input dimension

### ISSUE THREAD

**Issue by shiningrain** [keras-team/keras#13854]
## System information

- Have I written custom code (as opposed to using example directory):
- OS Platform and Distribution (e.g., Linux Ubuntu 16.04): Windows 10 & Linux Ubuntu 18.04
- Tensorflow backend (yes / no): yes
- Tensorflow version：1.15.0
- Cntk version: 2.7
- Theano version:1.04
- Keras version: 2.3.1
- Python version: 3.6.9
- CUDA/cuDNN version: -
- GPU model and memory: -

## Describe the current behavior
CNTK fails in building a model with the layer `ConvLSTM2D` or loading the same model built by Tensorflow/Theano (see the following table), with the error shown below (`cntk/ops/__init__.py line 421`). **For detailed parameters of `ConvLSTM2D`, you can refer to the following code snippet**.

``` ocaml
<class 'ValueError'>, ValueError('Convolution operation requires that kernel dim 5 <= input dim 4
```
|                | Build model       | Load Theano-built model |Load Tensorflow-built mode|
| :------------: | :---------------: | :---------------------: |:-:|
| **Tensorflow** |&radic;  | &radic;                 |&radic;   |
| **CNTK**       | &times; | &times;     |&times;   |
| **Theano**     | &radic;           | &radic;                 |&radic;   |

The table reveals that both Tensorflow and Theano succeed in building such a model, and get similar outputs, as shown in the following picture.  Notably, `diff3` in the picture means the sum of the absolute difference of output between the two backends. 
![pic22](https://user-images.githubusercontent.com/46860123/75794636-094caf00-5dac-11ea-8c8c-47464a745899.jpg)

### Analysis for CNTK error
We found this **CNTK error is jointly influenced by the input dimension and `kernel_size`**. The input dimension is calculated in `keras\backend\cntk_backend.py line 1731`, shown as the following picture.

![pic31](https://user-images.githubusercontent.com/46860123/75794665-15d10780-5dac-11ea-8363-cd8007ad6035.jpg)

Through the calculation in line 1731, different parameters will get different shapes, as shown in the following table.

`kernel_size=4; input_shape=[16,16,8]`

|strides|output shape after `convolution`|
|-|-|
|1|[8,13,5]|
|2|[8,7,3]|
|3|[8,5,2]|
|4|[8,4,2]|
|5|[8,3,1]|

The output shape after `convolution` operation will be used in `convolution` in `cntk\ops\__init__.py line 421`, shown as the following picture.

![pic34](https://user-images.githubusercontent.com/46860123/75794709-23868d00-5dac-11ea-8b91-ebec6175864b.jpg)

In this `convolution` process, the 3rd dimension value of `convolution_map` (e.g., 4 here) will compare with the 2nd dimension value of `toSequence_Input4`(e.g., 13). And the 4th dimension value of `convolution_map`(e.g., 4 here) will compare with the 3rd dimension value of `toSequence_Input4`(e.g., 5 here). **If the value in `convolution_map`is greater than the corresponding value in `toSequence_Input4`, CNTK will arise an error** like `ValueError('Convolution operation requires that kernel dim xx <= input dim xx`. 

As a example, if we set `strides=1` and `kernel_size=4 or (4,4)`, the shape of `toSequence_Input4` will be [8,13,5], as shown in the above picture. 4 is less than 5 or 13, and the error will not be triggered in this case.  Contrastly, If we change the `strides=3`,  the shape of `toSequence_Input4` will be [8,5.2]. 4 is greater than 2, and this leads to the error ` ValueError('Convolution operation requires that kernel dim 4 <= input dim 2`.

## Key insights
The implementation of  `ConvLSTM2D` on CNTK seems to have flaws. As summarized above, **when the value of `kernel_size` exceeds input dimension, CNTK will crash to build a model but other backends do not have this problem**.  

Additionally, **there is no relevant hints or warnings in the Keras document for this problem of inconsistent backend performance**, which may confuse the CNTK users.

## Application
When I run a MobileNetv2 model ( with ConvLSTM2D) on CNTK backend, it will arise the error shown in the following picture. However, this model can be normally predicted on Tensorflow and Theano, and get similar outputs.
![a823233a5c610c9e134716baeca692f](https://user-images.githubusercontent.com/46860123/75794761-3600c680-5dac-11ea-8279-b7ac5219b0e5.jpg)
![bff796aa9382f07a5f13564c2d983f4](https://user-images.githubusercontent.com/46860123/75794769-38632080-5dac-11ea-83b9-89d3a283a60c.png)
![73ef67f8780bc837554e905a89b93cc](https://user-images.githubusercontent.com/46860123/75794779-3ac57a80-5dac-11ea-83dc-56b4319465d2.png)

## Code to reproduce the issue
``` python
import os
import numpy as np
import keras.layers as L
import keras.backend as K
import importlib
from keras.models import load_model
from keras.engine import Model, Input

backends = ['cntk','tensorflow','theano']

def acc_abs_diff(output1, output2):
    assert output1.shape == output2.shape
    abs_diff = np.abs(output1-output2)
    return  np.sum(abs_diff)

def set_keras_backend(backend='tensorflow'):
    if K.backend() != backend:
        os.environ['KERAS_BACKEND'] = backend
        importlib.reload(K.load_backend)
        importlib.reload(K)
        assert K.backend() == backend


kwargs = {
    'filters': 8,
		'kernel_size': 5, 
    #'kernel_size': (1,5), 
		'strides':1,
		'padding': 'valid',
		'data_format': 'channels_first',
		'dilation_rate': 1,
		'use_bias': True,
		'unit_forget_bias': False,
		'return_sequences': True,
		'go_backwards': False,
		'stateful': False,
		'dropout': 0.7232577469807254,
		'recurrent_dropout': 0.7507926892266159
}

input_data = (10 * np.random.random((1,16,16,16,8)))
input = input_data.astype('float32')
set_keras_backend('cntk')
layer = L.convolutional_recurrent.ConvLSTM2D(**kwargs)
x = Input(batch_shape=input.shape)
y = layer(x)
bk_model = Model(x, y)
model_path = os.path.join('./', 'model.h5')
bk_model.save(model_path, bk_model)
output = {}
for bk in backends:
    try:
        set_keras_backend(backend=bk)
        model = load_model(model_path)
        output[bk] = model.predict(input)
    except:
        print('error result')
print('finish')
```

**Comment by Saduf2019** [comment]
@shiningrain 
We see that you are using old version of tensorflow which is not actively supported now, please verify on 2.x and open a new issue in case you face any errors, hence moving this to closed status.
